### PR TITLE
feat: added cart management

### PR DESCRIPTION
**Pull request by hiteshverma27**
- added add to cart
- added increase qty, decrease qty and remove from cart

### CODE REVIEW

File: src/components/CartProductCard/CartProductCard.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import { useCart } from "../../contexts";
+
+function CartProductCard({ item }) {
+  console.log(item);
```
Comment (anupkgurung) [inline src/components/CartProductCard/CartProductCard.js:4]: remove logs before push.

File: src/components/CartProductCard/CartProductCard.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  const { _id, catagoryName, title, price, qty } = item;
```
Comment (anupkgurung) [inline src/components/CartProductCard/CartProductCard.js:5]: suggestion : could have destructured the item in the parameter iteslef.

File: src/pages/products/Products.js [hunk 3]
```diff
@@ -1,49 +1,64 @@
...
+          filterDispatch({
+            type: "INITIAL_PRODUCTS",
+            payload: response.data.products,
+          });
+        })
+        .catch((err) => {
+          console.log(err);
+        });
+    };
+    getData();
```
Comment (anupkgurung) [inline src/pages/products/Products.js:28]: Why not use IIFE instead ? it would be much easier

**Review by anupkgurung — APPROVED** [review]
Left few suggestion.

File: src/components/CartProductCard/CartProductCard.js (hunk 1 above)
Comment (hiteshverma27) [reply src/components/CartProductCard/CartProductCard.js:4]: Will remove them, thanks for pointing out.

File: src/pages/products/Products.js (hunk 3 above)
Comment (hiteshverma27) [reply src/pages/products/Products.js:28]: Thanks for pointing out, I will learn about IIFE and implement in the project.

File: src/App.css [hunk 4]
```diff
@@ -1,5 +1,201 @@
-@import url("https://lightsout-store.netlify.app/styles/login.css");
+/* @import url("https://lightsout-store.netlify.app/styles/login.css");
 @import url("https://lightsout-store.netlify.app/style.css");
 @import url("https://lightsout-store.netlify.app/styles/product.css");
 @import url("https://lightsout-store.netlify.app/styles/wishlist.css");
```
Comment (Aditi0505) [inline src/App.css:4]: Remove these if not needed.

File: src/App.css [hunk 5]
```diff
@@ -1,5 +1,201 @@
...
+}
+.login-div {
+  align-items: center;
+  overflow-y: auto;
+  min-height: max-content;
+  background: rgba(255, 255, 255, 0.5);
+  backdrop-filter: blur(5px);
+}
+
+@media screen and (max-width: 425px) {
```
Comment (Aditi0505) [inline src/App.css:154]: Change this to em.

File: src/App.css [hunk 6]
```diff
@@ -1,5 +1,201 @@
...
+#menu-icon {
+  display: none;
+}
+#search-mobile {
+  display: none;
+}
+
+/* *********** Media queries for mobile ************* */
+
+@media screen and (max-width: 768px) {
```
Comment (Aditi0505) [inline src/App.css:95]: Convert this unit to em.

File: src/components/Header/Header.js [hunk 7]
```diff
@@ -1,15 +1,16 @@
...
       <input type="search" className="input w-100per m-auto flex-center-center" placeholder="Search LightsOut" id="search-mobile" />
       <img className="w-100per img-cover h-100vh " src="https://github.com/hiteshverma27/LightsOut-Store/blob/development/Assets/wallpaperflare.com_wallpaper%20(1).jpg?raw=true" alt="header-img" />
       <div className="center-div-method-2 header-text">
         <h1 className="w-100vw text-align-center">For the love of Formula 1</h1><br />
-        <a className="m-auto flex-center-center shop-now-btn" href="/Pages/product.html">
-          <button className="bg-accent p-2 mt-2 flex-center-center">Shop Now <span
-            className="material-icons icon-s3">arrow_forward_ios</span></button>
-        </a>
+        <div className="m-auto flex-center-center shop-now-btn" href="/Pages/product.html">
+          <Link to={"/products"} className="bg-accent p-2 mt-2 flex-center-center">Shop Now <span
```
Comment (Aditi0505) [inline src/components/Header/Header.js:11]: Let me know why are we using {} here?

File: src/components/Navigation/Navigation.js [hunk 8]
```diff
@@ -1,34 +1,60 @@
...
-        </ul>
-        <span className="material-icons icon-sm" id="menu-icon">menu</span>
-    </nav>;
+  const { cart } = useCart();
+  return (
+    <nav
+      className="w-100vw flex-space_between-center px-10 py-1 bg-white"
+      id="nav"
+    >
+      <Link to={"/"}>
```
Comment (Aditi0505) [inline src/components/Navigation/Navigation.js:12]: No need to provide {} here

File: src/components/Navigation/Navigation.js [hunk 9]
```diff
@@ -1,34 +1,60 @@
...
+            </span>
+            {cart.length ? (
+              <span className="badge badge-red">
+                {cart.length ? cart.length : null}
+              </span>
+            ) : null}
+          </Link>
+        </li>
+        <li className="mx-2">
+          <Link to={"/"} href="/Pages/wishlist.html">
```
Comment (Aditi0505) [inline src/components/Navigation/Navigation.js:34]: Remove href attribute.

File: src/components/Navigation/Navigation.js [hunk 10]
```diff
@@ -1,34 +1,60 @@
...
+    <nav
+      className="w-100vw flex-space_between-center px-10 py-1 bg-white"
+      id="nav"
+    >
+      <Link to={"/"}>
+        <h2>LightsOut</h2>
+      </Link>
+      <ul className="flex-space_between-center" id="nav-icons">
+        <li className="mx-2">
+          <Link to={"/"} href="/Pages/login.html">
```
Comment (Aditi0505) [inline src/components/Navigation/Navigation.js:17]: Remove href attribute.

**Review by Aditi0505 — APPROVED** [review]
Added few suggestions for you to look at.
Please add description and screenshots/gif and/or preview links for better visualization.
- Good folder structure
- Good naming convention followed

**Pull request merged**